frontend/www/js/omegaup/components/arena/RunSubmitPopup.vue

```diff
@@ -0,0 +1,323 @@
+      { language: 'cpp11-clang', name: 'C++11 (clang++ 10.0)' },
+      { language: 'cpp17-gcc', name: 'C++17 (g++ 9.3)' },
+      { language: 'cpp17-clang', name: 'C++17 (clang++ 10.0)' },
+      { language: 'java', name: 'Java (openjdk 14.0)' },
+      { language: 'py', name: 'Python 2.7' },
+      { language: 'py2', name: 'Python 2.7' },
+      { language: 'py3', name: 'Python 3.8' },
+      { language: 'rb', name: 'Ruby (2.7)' },
+      { language: 'cs', name: 'C# (8.0, dotnet 3.1)' },
+      { language: 'pas', name: 'Pascal (fpc 3.0)' },
+      { language: 'cat', name: 'Output Only' },
+      { language: 'hs', name: 'Haskell (ghc 8.6)' },
+      { language: 'lua', name: 'Lua (5.3)' },
+    ];
+
+    allLanguages
+      .filter(
+        (item) =>
+          this.languages.includes(item.language) || item.language === '',
+      )
+      .forEach((optionItem) => {
+        allowedLanguages[optionItem.language] = optionItem.name;
+      });
+    return allowedLanguages;
+  }
+
+  get extension(): string {
+    if (!this.selectedLanguage || this.selectedLanguage === 'cat') {
+      return '';
+    }
+    if (this.selectedLanguage.startsWith('cpp')) {
+      return '.cpp';
+    }
+    if (this.selectedLanguage.startsWith('c11-')) {
+      return '.c';
+    }
+    if (this.selectedLanguage.startsWith('py')) {
+      return '.py';
+    }
+    return `.${this.selectedLanguage}`;
+  }
+
+  @Watch('preferredLanguage')
+  onPreferredLanguageChanged(newValue: string): void {
+    this.selectedLanguage = newValue;
+  }
+
+  onSubmit(): void {
+    if (!this.canSubmit) {
+      alert(
+        ui.formatString(T.arenaRunSubmitWaitBetweenUploads, {
+          submissionGap: Math.ceil(
+            (this.nextSubmissionTimestamp.getTime() - Date.now()) / 1000,
+          ),
+        }),
+      );
+      return;
+    }
+
+    if (!this.selectedLanguage) {
+      alert(T.arenaRunSubmitMissingLanguage);
+      return;
+    }
+    const file = this.inputFile.files?.[0];
+    if (file) {
+      const reader = new FileReader();
+
+      reader.onload = (e) => {
+        const result = e.target?.result ?? null;
+        if (result === null) return;
+        this.$emit('submit-run', result as string, this.selectedLanguage);
+      };
+
+      const validExtensions = [
+        'cpp',
+        'c',
+        'cs',
+        'java',
+        'txt',
+        'hs',
+        'kp',
+        'kj',
+        'p',
+        'pas',
+        'py',
+        'rb',
+        'lua',
+      ];
+
+      if (
+        this.selectedLanguage !== 'cat' ||
+        file.type.indexOf('text/') === 0 ||
+        validExtensions.includes(this.extension)
+      ) {
+        if (this.inputLimit && file.size >= this.inputLimit) {
+          alert(
+            ui.formatString(T.arenaRunSubmitFilesize, {
+              limit: `${this.inputLimit / 1024} KiB`,
+            }),
+          );
+          return;
+        }
+        reader.readAsText(file, 'UTF-8');
+      } else {
```

Comment:
```suggestion
        reader.readAsText(file, 'UTF-8');
        return;
      }
```

aquí se puede usar el guard clause pattern!

Reply:
Listo.